### PR TITLE
define appr interface and wire with chart resource

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/1902

Connects appr service and chart resource. I've added also an interface to appr in order to have both components properly decoupled.

### CODE REVIEW

**Comment by rossf7** [comment]
Actually the mock is already there :) Nice!

**Comment by fgimenez** [comment]
> For me, the previous name makes more sense because we are getting the release that has been pushed to the specified channel.

Thanks a lot for pointing it out, previously the client was getting information about the package without including channel info, that's why the returned release was the default one, which led me to rename the method. I've changed it back the `GetRelease` and added channel info to the query, PTAL.

File: service/chartconfig/v1/appr/appr.go [hunk 1]
```diff
@@ -63,23 +64,26 @@ func New(config Config) (*Client, error) {
 }
 
 // GetRelease queries CNR for the release name of the chart represented by the
-// given custom object
+// given custom object (including channel info)
```
Comment (rossf7) [inline service/chartconfig/v1/appr/appr.go]: Nitpick, needs a `.`.

Comment (fgimenez) [reply service/chartconfig/v1/appr/appr.go]: Sure, done.

**Pull request merged**